StreamChatClient_v3/Controllers/ChannelController.swift

```diff
@@ -28,55 +67,199 @@ public class ChannelController<ExtraData: ExtraDataTypes>: Controller {
     
     /// Acts like the `NSFetchedResultsController`'s delegate and aggregates the reported changes into easily consumable form.
     private(set) lazy var changeAggregator: ChangeAggregator<ChannelDTO, ChannelModel<ExtraData>> = {
-        let aggregator: ChangeAggregator<ChannelDTO, ChannelModel<ExtraData>> = .init(itemCreator: ChannelModel<ExtraData>.create)
+        let aggregator: ChangeAggregator<ChannelDTO, ChannelModel<ExtraData>>
+            = self.environment.changeAggregatorBuilder(ChannelModel<ExtraData>.create)
         
         aggregator.onChange = { [unowned self] (_: [Change<ChannelModel<ExtraData>>]) in
             guard let channel = self.fetchResultsController.fetchedObjects?.first
                 .map(ChannelModel<ExtraData>.create(fromDTO:)) else { return }
-            
-            self.delegate?.channelController(self, didUpdateChannel: channel)
+            self.channel = channel
+            self.delegateCallback {
+                $0?.channelController(self, didUpdateChannel: channel)
+            }
         }
```

Comment:
The `ChangeAggregator` was designed for observing a list, so here, we'd ideally use it for observing the messages. We need to create something like `SingleEntityChangeAggregator`, which will provide a better API for observing a single entity.